packages/auth-construct/src/construct.ts

```diff
@@ -141,24 +146,41 @@ export class AmplifyAuth
     props: AuthProps = DEFAULTS.IF_NO_PROPS_PROVIDED
   ) {
     super(scope, id);
-
     this.name = props.name ?? '';
     this.domainPrefix = props.loginWith.externalProviders?.domainPrefix;
-
     // UserPool
     this.computedUserPoolProps = this.getUserPoolProps(props);
+
+    const userPoolProps = {
+      ...this.computedUserPoolProps,
+    };
+    let customSenderKmsKey: Key | undefined;
+    if (
+      props.senders?.email &&
+      props.senders.email instanceof lambda.Function
+    ) {
+      if (!customSenderKmsKey) {
+        customSenderKmsKey = new Key(this, 'CustomSenderKey', {
+          enableKeyRotation: true,
+        });
+      }
+      customSenderKmsKey.grantDecrypt(props.senders.email);
+      props.senders.email.addPermission('CognitoInvokeEmail', {
```

Comment:
```suggestion
      props.senders.email.addPermission('CognitoInvokeEmailSenderHandler', {
```

or something more descriptive.